mlflow/models/wheeled_model.py

```diff
@@ -0,0 +1,158 @@
+        :param model_uri: registered model uri of the form
+                            models:/<model_name>/<model_version/stage/latest>
+        :param registered_model_name: If given, create a model version under
+                                  ``registered_model_name``, also creating a registered model if one
+                                  with the given name does not exist.
+        """
+        model_name, _, _ = _parse_model_uri(model_uri)
+        return Model.log(
+            artifact_path=artifact_path,
+            flavor=WheeledModel(model_uri),
+            registered_model_name=registered_model_name or model_name,
+        )
+
+    def save_model(self, path, mlflow_model=None):
+        """
+        Given an existing registered model, saves the model along with it's dependencies stored as
+        wheels to a path on the local file system.
+
+        This does not modify existing model behavior or
+        existing model flavors. It simply downloads the model dependencies as wheels and modifies
+        the requirements.txt and conda.yaml file to point to the downloaded wheels.
+        :param path: Local path where the model is to be saved.
+        :param mlflow_model: The new :py:mod:`mlflow.models.Model` metadata file to store the
+                            updated model metadata.
+        """
+        from mlflow.pyfunc import FLAVOR_NAME, ENV
+
+        path = os.path.abspath(path)
+        _validate_and_prepare_target_save_path(path)
+
+        local_model_path = _download_artifact_from_uri(self._model_uri, output_path=path)
+
+        wheels_dir = os.path.join(local_model_path, _WHEELS_FOLDER_NAME)
+        pip_requirements_path = os.path.join(local_model_path, _REQUIREMENTS_FILE_NAME)
+        model_metadata_path = os.path.join(local_model_path, MLMODEL_FILE_NAME)
+
+        model_metadata = Model.load(model_metadata_path)
+
+        # Check if the model file has `wheels` set to True
+        if model_metadata.__dict__.get(_WHEELS_FOLDER_NAME, None):
+            raise MlflowException("Cannot add wheels to a wheeled model", BAD_REQUEST)
+
+        conda_env = model_metadata.flavors.get(FLAVOR_NAME, {}).get(ENV, None)
+        conda_env_path = os.path.join(local_model_path, conda_env)
+        if conda_env is None:
+            raise MlflowException(
+                "Can not add wheels for model with no conda environment.", BAD_REQUEST
+            )
+        if not os.path.isfile(pip_requirements_path):
+            raise Exception("Can not add wheels for model with no 'requirements.txt'.")
+
+        self._download_wheels(dst_path=wheels_dir, pip_requirements_path=pip_requirements_path)
+
+        # Update requirements.txt with wheels
+        pip_wheels = self._overwrite_pip_requirements_with_wheels(
+            wheels_dir=wheels_dir, pip_requirements_path=pip_requirements_path
+        )
+
+        _mlflow_conda_env(path=conda_env_path, additional_pip_deps=pip_wheels, install_mlflow=False)
```

Comment:
Would this overwrite the whole conda file? I think want to change only the pip requirements section - i.e. we want to preserve channels and python version and any conda packages.